### PR TITLE
Exit addEarplugs if earplugs are used

### PR DESCRIPTION
**Pull request by jonpas**
This solves duplicating Earplugs in persistence scenarios (eg. persistence mods). It will still add the earplugs if none are equipped, but in terms of persistence you would normally equip them and be done with it anyways.

### CODE REVIEW

File: addons/hearing/functions/fnc_addEarPlugs.sqf [hunk 1]
```diff
@@ -15,9 +15,12 @@
  */
 #include "script_component.hpp"
 
-private ["_unit", "_launcher"];
+PARAMS_1(_unit);
 
-_unit = _this select 0;
+// Exit if soldier has earplugs already in (persistence scenarios)
+if (FUNC(hasEarPlugsIn)) exitWith {};
```
Comment (PabstMirror) [inline addons/hearing/functions/fnc_addEarPlugs.sqf]: "Error if: Type code, expected Bool"
probably should be:
`if ([_unit] call FUNC(hasEarPlugsIn)) exitWith {};`

Comment (jonpas) [inline addons/hearing/functions/fnc_addEarPlugs.sqf]: Oh crap, forgot to save the file! :grin:

**Comment by alganthe** [comment]
wouldn't removing earplugs on respawn be a better solution ?

**Comment by jonpas** [comment]
> wouldn't removing earplugs on respawn be a better solution ?

You mean by a mission/persistence system used? Yes of course, but this is not the only point of the PR. Right now if you have earplugs in already it will add another, which you'd normally just throw on the ground.

**Comment by jonpas** [comment]
Would it be ok to add an ACE_Settings setting to disable that automatic addition?

**Comment by esteldunedain** [comment]
@jonpas, please resolve the conflict generated by #1763. Besides that I think this is ready to be merged.

**Comment by jonpas** [comment]
Resolved conflicts and combined both PRs for performance.

**Comment by esteldunedain** [comment]
Thanks!

**Pull request merged**